test/e2e/adminapi_cluster_getlogs.go

```diff
@@ -59,24 +56,12 @@ func testGetPodLogsOK(ctx context.Context, containerName, podName, namespace str
 	}()
 
 	By("waiting for the pod to successfully terminate")
-	err = wait.PollInfinite(time.Second*5, func() (done bool, err error) {
+	Eventually(func(g Gomega, ctx context.Context) {
 		pod, err = clients.Kubernetes.CoreV1().Pods(namespace).Get(ctx, pod.Name, metav1.GetOptions{})
-		Expect(err).NotTo(HaveOccurred())
+		g.Expect(err).NotTo(HaveOccurred())
 
-		switch pod.Status.Phase {
-		case corev1.PodSucceeded:
-			return true, nil
-		case corev1.PodPending:
-			if pod.CreationTimestamp.Time.Add(5*time.Minute).Unix() < time.Now().Unix() {
-				return false, errors.New("pod was pending for more than 5min")
```

Comment:
Do we not care about the "takes more than 5 mins to schedule pod" case?

Reply:
It is covered on line 63:

```go
g.Expect(pod.Status.Phase).To(Equal(corev1.PodSucceeded))))
```

`Eventually` has a timeout (5 minutes, [default](https://github.com/Azure/ARO-RP/blob/180eb992fce1e2a132e7643a31edc4c3c04d3bd4/test/e2e/setup.go#L436)). So if we exeed the timeout due to not satisfying this matcher - we will see it in the failure message produced by line 63. It will say that we expected phase `Succeeded`, but got `Pending` (or something else if there was another reason).

Reply:
Makes sense. Thanks!